core/src/toga/window.py

```diff
@@ -459,21 +466,29 @@ def visible(self, visible: bool) -> None:
     ######################################################################
 
     @property
-    def full_screen(self) -> bool:
-        """Is the window in full screen mode?
-
-        Full screen mode is *not* the same as "maximized". A full screen window
-        has no title bar, toolbar or window controls; some or all of these
-        items may be visible on a maximized window. A good example of "full screen"
-        mode is a slideshow app in presentation mode - the only visible content is
-        the slide.
-        """
-        return self._is_full_screen
-
-    @full_screen.setter
-    def full_screen(self, is_full_screen: bool) -> None:
-        self._is_full_screen = is_full_screen
-        self._impl.set_full_screen(is_full_screen)
+    def state(self) -> WindowState:
+        """The current state of the window."""
+        return self._impl.get_window_state()
+
+    @state.setter
+    def state(self, state: WindowState) -> None:
+        current_state = self._impl.get_window_state()
+        if current_state != state:
+            if not self.resizable and state in {
+                WindowState.MAXIMIZED,
+                WindowState.FULLSCREEN,
+                WindowState.PRESENTATION,
+            }:
+                warnings.warn(
+                    f"Cannot set window state to {state} of a non-resizable window."
+                )
+            else:
+                # Changing the window state while the app is in presentation mode
+                # can cause rendering glitches. Hence, first exit presentation and
+                # then set the new window state.
+                self.app.exit_presentation_mode()
```

Comment:
What about if the app is already *in* presentation mode?

Reply:
Then also the presentation mode should be exited, as on platforms other than macOS, presentation mode is window based and not app based. So, if the user sets `window1.state = WindowState.PRESENTATION`, while the app is in PRESENTATION mode with multiple windows, then it first exits the presentation mode and then sets only the specified window to PRESENTATION.